### DISCUSSION TITLE
What is the documentation comment ID for function pointers?

### DISCUSSION THREAD

**Discussion by svick** [dotnet/csharplang#3978] [Q&A] [answered]
The C# 6.0 spec [describes the format of documentation comment ID](https://docs.microsoft.com/en-us/dotnet/csharp/language-reference/language-specification/documentation-comments#id-string-format) for various kinds of types. Of course it doesn't describe the format for C# 9.0 function pointers, but I didn't find that format specified in [the function pointers proposal](https://docs.microsoft.com/en-us/dotnet/csharp/language-reference/proposals/csharp-9.0/function-pointers) or anywhere else.

Even with latest daily build of Roslyn, there doesn't seem to be any format specified. For example, consider the following code:

```c#
string code = @"
class C
{
    unsafe void M(delegate*<int, void> p, int i) {}
}
";

var tree = SyntaxFactory.ParseSyntaxTree(code);

var compilation = CSharpCompilation.Create(null, options: new CSharpCompilationOptions(OutputKind.DynamicallyLinkedLibrary, allowUnsafe: true))
	.AddSyntaxTrees(tree)
	.AddReferences(MetadataReference.CreateFromFile(typeof(object).Assembly.Location));
	
var method = compilation.GetSymbolsWithName("M").Single();
Console.WriteLine(DocumentationCommentId.CreateDeclarationId(method));
Console.WriteLine(method.GetDocumentationCommentId());
```

This prints:

```
M:C.M(,System.Int32)
M:C.M(,System.Int32)
```

which shows empty string as the documentation comment ID for a function pointer parameter.

Should documentation comment ID for function pointers be specified by the language? Or this a question for the compiler?

Also, the same issue might exist for `dynamic` (which has been previously reported at https://github.com/dotnet/roslyn/issues/34839).

**Comment by 333fred** [comment, marked as answer]
This is a question for the compiler, though the spec should also be updated.

**Reply by svick** [reply]
In that case, can you (or someone with the rights) transfer this discussion to dotnet/roslyn? Or should I close this here and open a new discussion there?

**Reply by 333fred** [reply]
We can't convert discussions to issues, so this should be a new issue on the roslyn side :)

**Reply by svick** [reply]
@333fred Ok, opened https://github.com/dotnet/roslyn/issues/48363.
